Sources/Tracking/CustomerIO.swift

```diff
@@ -238,6 +246,59 @@ public class CustomerIO: CustomerIOInstance {
             )
     }
 
+    // Swizzled method for APN device token.
+    @objc
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        let token = String(apnDeviceToken: deviceToken)
+        Self.shared.registerDeviceToken(token)
+    }
+
+    // Implement method swizzling for APN device token. Swizzling will ensure that the
+    // device token gets registered without needing the developer to add call respective
+    // Customer.io SDK methods to register the device.
+    @available(iOSApplicationExtension, unavailable)
+    func swizzleDidReceiveRemoteNotification() {
+        DispatchQueue.main.async {
+            let appDelegate = UIApplication.shared.delegate
+            let appDelegateClass: AnyClass? = object_getClass(appDelegate)
+            let originalSelector = #selector(UIApplicationDelegate.application(_:didRegisterForRemoteNotificationsWithDeviceToken:))
+            let swizzledSelector = #selector(self.application(_:didRegisterForRemoteNotificationsWithDeviceToken:))
+            guard let swizzledMethod = class_getInstanceMethod(CustomerIO.self, swizzledSelector) else {
+                return
+            }
+            if let originalMethod = class_getInstanceMethod(appDelegateClass, originalSelector) {
+                method_exchangeImplementations(originalMethod, swizzledMethod)
+            }
```

Comment:
The `CustomerIO` class [has an existing function to swizzle methods](https://github.com/customerio/customerio-ios/blob/f304189cc5504ac763c3d69307da783213fe903d/Sources/Tracking/CustomerIOImplementation%2BScreenViews.swift#L11-L27) that can be re-used here.

Reply:
https://github.com/customerio/customerio-ios/pull/378#discussion_r1314728612